File: data/src/main/java/com/icehousecorp/maunorafiq/data/weather/entity/mapper/WeatherEntityDataMapper.java
package com.icehousecorp.maunorafiq.data.weather.entity.mapper;

import javax.inject.Inject;
import javax.inject.Singleton;

import com.icehousecorp.maunorafiq.data.weather.entity.response.WeatherResponse;
import com.icehousecorp.maunorafiq.domain.weather.Weather;

import static com.icehousecorp.maunorafiq.data.Constant.iconImageUrl;

/**
 * Created by Raffi on 11/25/2016.
 */
@Singleton
public class WeatherEntityDataMapper {

    @Inject
    public WeatherEntityDataMapper(){}

    public Weather transform(WeatherResponse weatherResponse, String city) {
        Weather weather = null;
        if (weatherResponse != null) {
            weather = new Weather(weatherResponse.getId());
            weather.setCityName(city);
            weather.setWeatherId(weatherResponse.getWeather().get(0).getId());
            weather.setWeatherName(weatherResponse.getWeather().get(0).getMain());
            weather.setWeatherDescription(weatherResponse.getWeather().get(0).getDescription());
            weather.setWeatherIcon(iconImageUrl + weatherResponse.getWeather().get(0).getIcon() + ".png");
            weather.setUtcTime(weatherResponse.getDt());
            weather.setTemperature(Integer.toString(weatherResponse.getMain().getTemp().intValue()));
            weather.setPressure(weatherResponse.getMain().getPressure().toString());
            weather.setHumidity(weatherResponse.getMain().getHumidity().toString());
        }
        return weather;
    }
}

File: data/src/main/java/com/icehousecorp/maunorafiq/data/weather/entity/mapper/WeatherEntityDataMapper.kt
package com.icehousecorp.maunorafiq.data.weather.entity.mapper

import javax.inject.Inject
import javax.inject.Singleton

import com.icehousecorp.maunorafiq.data.weather.entity.response.WeatherResponse
import com.icehousecorp.maunorafiq.domain.weather.Weather

import com.icehousecorp.maunorafiq.data.Constant.iconImageUrl

/**
 * Created by Raffi on 11/25/2016.
 */
@Singleton
class WeatherEntityDataMapper
@Inject
constructor() {

    fun transform(weatherResponse: WeatherResponse?, city: String): Weather? {
        var weather: Weather? = null
        if (weatherResponse != null) {
            weather = Weather(weatherResponse.id!!)
            weather.cityName = city
            weather.weatherId = weatherResponse.weather[0].id!!
            weather.weatherName = weatherResponse.weather[0].main
            weather.weatherDescription = weatherResponse.weather[0].description
            weather.weatherIcon = iconImageUrl + weatherResponse.weather[0].icon + ".png"
            weather.utcTime = weatherResponse.dt!!
            weather.temperature = Integer.toString(weatherResponse.main!!.temp!!.toInt())
            weather.pressure = weatherResponse.main!!.pressure!!.toString()
            weather.humidity = weatherResponse.main!!.humidity!!.toString()
        }
        return weather
    }
}
